modules/core/src/lib/layers/selection-layer.js

```diff
@@ -0,0 +1,272 @@
+// @flow
+/* eslint-env browser */
+import { PolygonLayer, COORDINATE_SYSTEM } from 'deck.gl';
+import { point, polygon } from '@turf/helpers';
+import turfBbox from '@turf/bbox';
+import turfBboxPolygon from '@turf/bbox-polygon';
+import turfBuffer from '@turf/buffer';
+import turfDifference from '@turf/difference';
+
+import { SELECTION_TYPE } from '../deck-renderer/deck-drawer';
+import type {
+  ClickEvent,
+  StartDraggingEvent,
+  StopDraggingEvent,
+  PointerMoveEvent,
+  DoubleClickEvent
+} from '../event-types.js';
+
+import EditableLayer from './editable-layer.js';
+
+const defaultProps = {
+  selectionType: SELECTION_TYPE.RECTANGLE,
+  layerIds: [],
+  onSelection: () => {}
+};
+
+const POLYGON_LINE_COLOR = [0, 255, 0, 255];
+const POLYGON_FILL_COLOR = [255, 255, 255, 90];
+const POLYGON_LINE_WIDTH = 2;
+const POLYGON_DASHES = [20, 20];
+const POLYGON_THRESHOLD = 0.01;
+const EXPANSION_KM = 10;
+const LAYER_ID_VIEW = 'DeckDrawerView';
+const LAYER_ID_PICK = 'DeckDrawerPick';
+
+export default class SelectionLayer extends EditableLayer {
```

Comment:
I'm not sure I understand the point of this layer. Is it actually supposed to select objects in the `EditableGeoJsonLayer`? It doesn't seem to work. Also, I don't know that you need a whole separate layer class. I think you could make this just a specialized version of `EditableGeoJsonLayer` with custom styling and/or custom `ModeHandler`s. "Rect" select mode is just `drawRectangle` mode and "Lasso" select mode is just `drawPolygon` mode.

Reply:
The idea is to have a light-weigh selection logic.

Reply:
I also considered extending `EditableGeoJsonLayer`.

Reply:
Done.